### PR TITLE
Introduce datapusher.

### PR DESCRIPTION
**Pull request by zelima**
- Include Dockerfile and related configurations
- Include in docker compose and link to CKAN
- Remove hardcoded datapusher URL and make it variable in the config
- Include datapusher URL in secrets

Note: Datapusher is **not** must present service for the default setup.
However, it's present in plugins (for V.S. at least). So lack of it leads to errors

### CODE REVIEW

File: datapusher/setup/datapusher_settings.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+TESTING = False
+SECRET_KEY = str(uuid.uuid4())
+USERNAME = str(uuid.uuid4())
+PASSWORD = str(uuid.uuid4())
+
+NAME = 'datapusher'
+
+# database
+
+SQLALCHEMY_DATABASE_URI = 'sqlite:////tmp/job_store.db'
```
Comment (victor-nitu) [inline datapusher/setup/datapusher_settings.py:13]: Should we connect it to `db` service instead? Just a question, I'm not sure it adds any benefits...

File: datapusher/setup/wsgi.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import os
```
Comment (victor-nitu) [inline datapusher/setup/wsgi.py]: Not used, needs to go.

File: datapusher/setup/wsgi.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+import sys
```
Comment (victor-nitu) [inline datapusher/setup/wsgi.py]: Not used, needs to go.

File: docker-compose.yaml [hunk 4]
```diff
@@ -82,10 +82,20 @@ services:
...
+    ports:
+      - "8800:8800"
+    networks:
+    - ckan-multi
+
   ckan:
     depends_on:
     - redis
     - solr
+    - datapusher
```
Comment (victor-nitu) [inline docker-compose.yaml]: `ckan` should not depend on `datapusher` globally, i.e. for all instances. The `depends_on` is an array that cannot be overwritten in override files, so if you put it here we're stuck with it for all themes.

If you define it as dependency in the override files it's enough, the override files append to this array but cannot replace it. (also valid for all docker-compose array types)

File: .docker-compose.vital-strategies-theme.yaml [hunk 5]
```diff
@@ -52,3 +56,7 @@ services:
                                    CREATE EXTENSION IF NOT EXISTS postgis_topology;         \
                                    CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;            \
                                    CREATE EXTENSION IF NOT EXISTS postgis_tiger_geocoder;"
+  datapusher:
+    container_name: datapusher
+    build:
```
Comment (victor-nitu) [inline .docker-compose.vital-strategies-theme.yaml]: This was already defined in the main `docker-compose.yaml`. Remove it from here, as it's redundant.

File: docker-compose.yaml [hunk 6]
```diff
@@ -82,10 +82,20 @@ services:
     networks:
     - ckan-multi
 
+  datapusher:
+    container_name: datapusher
```
Comment (victor-nitu) [inline docker-compose.yaml]: I'm not sure, but this should probably go into the override. Or if it stays here, it must be removed from there.

File: .docker-compose.vital-strategies-theme.yaml [hunk 7]
```diff
@@ -52,3 +56,7 @@ services:
                                    CREATE EXTENSION IF NOT EXISTS postgis_topology;         \
                                    CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;            \
                                    CREATE EXTENSION IF NOT EXISTS postgis_tiger_geocoder;"
+  datapusher:
+    container_name: datapusher
```
Comment (victor-nitu) [inline .docker-compose.vital-strategies-theme.yaml:60]: This is already defined in main compose file. As the entire service is. Having its definition copied here is redundant, unless anything changes or adds to it.

**Review by victor-nitu — CHANGES_REQUESTED** [review]
Some YAML changes and we're good to go! :v:

File: datapusher/setup/datapusher_settings.py (hunk 1 above)
Comment (zelima) [reply datapusher/setup/datapusher_settings.py:13]: This is the variable in base image. It's later overridden `SQLALCHEMY_URL` I think. But I'm not sure neither tbh

File: docker-compose.yaml (hunk 4 above)
Comment (zelima) [reply docker-compose.yaml]: Hope you're sure about that. I don't exactly remember why I included here

**Review by victor-nitu — APPROVED** [review]

**Pull request merged**